### ISSUE TITLE
Upgrade to D9

### ISSUE THREAD

**Issue by thinkcollege** [communityinclusion/escoach#79]
Hi @ultimike I am converting my D8 sites to D9.  This is not the walk in the park advertised by the project, but I have figured it out in most cases, having to throw some useful/critical contrib modules overboard.  I am working on the ES Coach site locally.  Can you make sure your custom modules are 9-ready on the devel branch, and I will merge them into the live site branch?  Not sure where you are with remaining hours on the contract.  We should check in on what remaining tasks we need to wrap up to complete the project.

**Comment by thinkcollege** [comment]
Hi @ultimike The live site, es-coach.org is now running D9.  It went pretty smoothly.
Not sure what we want to do with rp-escoach.communityinclusion.org, the devel site.  I am kind of exhausted from some other sites that did not go so well, so I am in no hurry to convert it.  If you want that site in D9 we can devise a plan for that.

**Comment by ultimike** [comment]
Isn't es-coach.org just the `rack` branch and rp-escoach the `devel` branch? You don't want to merge for some reason? As we're still making commits to `devel`, at some point you're going to have to merge, right? Or are you just cherry-picking?

-mike

**Comment by thinkcollege** [comment]
I could do a rack > devel merge, but that would break the database on devel, unless someone went through and made sure extensions, etc. were prepared.  It would be easier to merge the code and wipe the devel site database, replacing it with the live site db.  Then there would be some brief fine tuning to get rid of the live site users and make sure the site was not sending out emails and so on.  I guess that needs to be done before I merge anything devel > live.

**Comment by ultimike** [comment]
I'm going to leave it up to you - either way, I don't think it will really affect what we're doing.

-mike

**Comment by thinkcollege** [comment]
I will probably do what I suggested above, but will check in with you before I push the button so you don't lose any code or db settings on devel.

**Comment by thinkcollege** [comment]
I may try just merging the code locally for kicks and see how badly (or not) it fuzzles the database.

**Comment by thinkcollege** [comment]
@ultimike @mgcobblestone  Okay, I was able to do the upgrade to D9 locally without wiping the database on rp-escoach.communityinclusion.org   1) I pulled latest commits to devel from yesterday and merged them into the rack branch 2) overwrote the devel branch with the rack branch and 3) ran the site on the devel site database locally after having deactivated some obsolete or troublesome contrib modules.  I am about to push the new code to the devel branch and have it deploy to rp-escoach.communityinclusion.org.  Okay with you?

**Comment by ultimike** [comment]
@thinkcollege That all sounds good, except that it seems like you're either going to have to tell us which "obsolete or troublesome contrib" modules we should uninstall first - or just send us a copy of the future D9 rp-escoach database.

Make sense?

thanks,  
-mike

**Comment by thinkcollege** [comment]
Okay, updated devel branch and rp-escoach.communityinclusion.org site.
I disabled the two FAPI modules and the Force Password Change module and that did it.  If you pull to your local after disabling those you should be good. Unless...whatever.  I can send you the database if necessary.

**Comment by ultimike** [comment]
@mgcobblestone ☝️

**Comment by ultimike** [comment]
@thinkcollege @mgcobblestone My local is all up-to-date - here's what I did:

1.  Fired up my local.  
2. Uninstalled "FAPI validation", "FAPI validation example form", and "Force password change" modules.  
3. I did a `git fetch` then a `git rebase origin/devel` - I got merge conflicts in survey_dashboard.js and QueryBuilder.php. 
4. I did a `git rebase --skip`, then a `git reset HEAD --hard`.  
5. @thinkcollege, it looks like maybe you didn't update the `devel` branch before doing the merge because at this point I have two commits on my `devel` branch that aren't pushed up. 
6. I did a `drush cr`. 
7. I did a `drush updb` - but it failed at first due to the missing "kint" module. I had to very carefully edit the `core.extension` entry in my local `config` database table to remove it. After that, I was able to run `drush updb`.
7. I updated my local to use PHP 7.4 (to match the change in Composer.json).  
8. I'm getting a lot of messages like the following: `Testing... function /var/www/html/web/modules/contrib/workflow/modules/workflow_devel/workflow_devel.module::workflow_devel_entity_update/172 = update > migration` - @thinkcollege Is this some debugging code of yours somewhere? It looks like it is coming from the Workflow Devel module - I disabled it on my local.  
9. I just pushed the two commits mentioned above to the `devel` branch on origin. I don't believe this undid anything, but maybe it pays for @mgcobblestone to double-check me?

-mike

**Comment by thinkcollege** [comment]
Sorry this was so painful.  I did check for the latest devel commits a few minutes before I merged into the rack branch.   If you go with your latest version of your custom modules things should be okay.  And things seemed to be running okay on rp-escoach.communityinclusion.org.
Yes, please deactivate that workflow devel module.  It is annoying.  I have found it mysteriously getting activated on a few different D8/9 sites.  I will remove it at some point.
